### PR TITLE
[POAE7-2596]move codegen to Expr

### PR DESCRIPTION
**Pull request by Ma-Jian1**
### What changes were proposed in this pull request?
move codegen from outside into Expr, dispatch codegen() by virtual method.

### Why are the changes needed?
dispatch by dynamic_cast is slow

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### Which label does this PR belong to?
INFRA

### CODE REVIEW

File: cider/type/plan/Analyzer2.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+ */
+#include "exec/nextgen/jitlib/JITLib.h"
+#include "exec/nextgen/jitlib/base/JITFunction.h"
+#include "exec/template/Execute.h"
+#include "type/plan/Analyzer.h"
+
+namespace Analyzer {
+using namespace cider::jitlib;
+
+JITTypeTag Expr::getJITTag(const SQLTypes& st) {
```
Comment (bigPYJ1151) [inline cider/type/plan/Analyzer2.cpp:29]: Maybe we could separate this func to ```nextgen/utils```? So other components could use it.

File: cider/type/plan/Analyzer.h [hunk 2]
```diff
@@ -323,6 +334,9 @@ class ColumnVar : public Expr {
   bool operator==(const Expr& rhs) const override;
   std::string toString() const override;
 
+ public:
+  JITExprValue& codegen(JITFunction& func) override;
```
Comment (bigPYJ1151) [inline cider/type/plan/Analyzer.h:338]: Should we move these ```Expr``` subclasses to separate files step-by-step? Maybe ```Analyzer.h``` is oversized.

File: cider/type/plan/Analyzer2.cpp (hunk 1 above)
Comment (Ma-Jian1) [reply cider/type/plan/Analyzer2.cpp:29]: only Expr needs this so far. 
We can make this change any time when we found others need this.

File: cider/type/plan/Analyzer.h (hunk 2 above)
Comment (Ma-Jian1) [reply cider/type/plan/Analyzer.h:338]: good idea

File: ci/scripts/intel_copyright_files.txt [hunk 3]
```diff
@@ -22,6 +22,10 @@ cider/exec/plan/parser/*Visitor*
 cider/exec/plan/parser/Substrait*
 cider/exec/plan/parser/ConverterHelper.*
 cider/exec/nextgen/*
+cider/type/plan/BinaryExpr.cpp
+cider/type/plan/ColumnExpr.cpp
+cider/type/plan/ConstantExpr.cpp
+cider/type/plan/Expr.cpp
```
Comment (bigPYJ1151) [inline ci/scripts/intel_copyright_files.txt:28]: Use regex?

Comment (Ma-Jian1) [reply ci/scripts/intel_copyright_files.txt]: done

**Review by bigPYJ1151 — APPROVED** [review]

File: cider/type/plan/BinaryExpr.cpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "BinaryExpr.h"
+#include "exec/template/Execute.h"  // for is_unnest
+
+namespace Analyzer {
+using namespace cider::jitlib;
+
+JITExprValue& BinOper::codegen(JITFunction& func) {
```
Comment (xieqi) [inline cider/type/plan/BinaryExpr.cpp:27]: These files are related to codegen, and it is not appropriate to put them in the plan directory

**Review by winningsix — APPROVED** [review]

File: cider/type/plan/BinaryExpr.cpp (hunk 4 above)
Comment (Ma-Jian1) [reply cider/type/plan/BinaryExpr.cpp:27]: then should we move all Analyzer out of plan?

Comment (winningsix) [reply cider/type/plan/BinaryExpr.cpp:27]: Hmm, will be better to decouple plan definition and codegen logic? Saying ```ColumnVar``` with ```ColumnVarCodegen```. Thoughts?

Comment (Ma-Jian1) [reply cider/type/plan/BinaryExpr.cpp:27]: we may reconsider how to refactor this API after the nextgen is finished.

**Pull request merged**